### PR TITLE
Add deployment scripts and documentation for the indexer

### PR DESCRIPTION
**Pull request by kadircet**

### CODE REVIEW

File: indexer/args.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (sam-mccall) [inline indexer/args.sh:1]: nit: no shebang, this is intended to be sourced?

File: indexer/rollout_new_release.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+source args.sh
+
+set -ex
+
+# Start by creating a new image.
+. push_new_docker_image.sh $@
```
Comment (sam-mccall) [inline indexer/rollout_new_release.sh:7]: any reason to source this rather than run it?

File: indexer/push_new_docker_image.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+cp Dockerfile "$TEMP_DIR/"
+cp cronjob "$TEMP_DIR/"
+cp run.sh "$TEMP_DIR/"
+cp ../download_latest_release_assets.py "$TEMP_DIR/"
+cp prepare.sh "$TEMP_DIR/"
+cp entry_point.sh "$TEMP_DIR/"
+cd "$TEMP_DIR"
+
+# Build the image, tag it for GCR and push.
+docker build --build-arg TOKEN="$TOKEN" -t "$IMAGE_IN_GCR" .
```
Comment (sam-mccall) [inline indexer/push_new_docker_image.sh:23]: we're not catching failures of these commands, maybe change the shebang to include `-e` (and remove the "bash" from the instructions)

**Review by sam-mccall — APPROVED** [review]
Thanks!

File: indexer/rollout_new_release.sh (hunk 2 above)
Comment (kadircet) [reply indexer/rollout_new_release.sh:7]: well, it actually ensures we're printing the correct usage message as this way we preserve $0 (and set -ex) from outer call.

File: indexer/push_new_docker_image.sh (hunk 3 above)
Comment (kadircet) [reply indexer/push_new_docker_image.sh:23]: got rid of shebang lines and added `set -ex`

**Pull request merged**